mixer/pkg/protobuf/yaml/dynamic/auth.go

```diff
@@ -0,0 +1,270 @@
+// Copyright 2018 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dynamic
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"time"
+
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/oauth"
+
+	policypb "istio.io/api/policy/v1beta1"
+	"istio.io/istio/pkg/log"
+)
+
+const (
+	// file path to the public CA certs
+	publicCACertsFile = "/etc/ssl/certs/ca-certificates.crt"
+
+	// whitelisted subjectAltName for MTLS identity check
+	// TODO(bianpengyuan): make this configurable
+	whitelisteSAN = "spiffe://cluster.local/ns/istio-system/sa/istio-mixer-service-account"
```

Comment:
Can we please use the function from pilot/pkg/proxy/envoy/infra_auth.go to at least keep track of the hard coded SAN? It seems like namespace is the only configurable part.

Reply:
Instead of hardcode, change the logic to use SAN parsed from mixer's client cert. We can make it more configurable in the future if needed.